File: app/src/main/java/com/digicoffer/lauditor/CommonFiles/CacheUtils/NetworkUtils.java
package com.digicoffer.lauditor.CommonFiles.CacheUtils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class NetworkUtils {

    public static boolean isInternetAvailable(Context context) {
        try {
            ConnectivityManager cm =
                    (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);

            if (cm == null) return false;

            NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
            return activeNetwork != null && activeNetwork.isConnected();

        } catch (Exception e) {
            return false;
        }
    }
}

File: app/src/main/java/com/digicoffer/lauditor/CommonFiles/CacheUtils/NetworkUtils.kt
package com.digicoffer.lauditor.CommonFiles.CacheUtils

import android.content.Context
import android.net.ConnectivityManager

class NetworkUtils {
    companion object {
        @JvmStatic
        fun isInternetAvailable(context: Context?): Boolean {
            if (context == null) return false
            return try {
                val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as? ConnectivityManager
                val activeNetwork = cm?.activeNetworkInfo
                activeNetwork != null && activeNetwork.isConnected
            } catch (e: Exception) {
                false
            }
        }
    }
}
